From bf676a5f1dde1f21bcd8e276ac21f6e2b86b661b Mon Sep 17 00:00:00 2001
From: Gabriel Potter <gabriel@potter.fr>
Date: Tue, 12 Jun 2018 12:04:19 +0200
Subject: [PATCH] Fix autogen_shortcuts.py for prompt_toolkit 2.0

---
 docs/autogen_shortcuts.py | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/docs/autogen_shortcuts.py b/docs/autogen_shortcuts.py
index 95c8d608b5..8535f1c285 100755
--- a/docs/autogen_shortcuts.py
+++ b/docs/autogen_shortcuts.py
@@ -1,7 +1,6 @@
 from os.path import abspath, dirname, join
 
-from IPython.terminal.interactiveshell import KeyBindingManager
-
+from IPython.terminal.shortcuts import create_ipython_shortcuts
 
 def name(c):
     s = c.__class__.__name__
@@ -42,8 +41,14 @@ def multi_filter_str(flt):
 log_filters = {'_AndList': 'And', '_OrList': 'Or'}
 log_invert =  {'_Invert'}
 
-kbm = KeyBindingManager.for_prompt()
-ipy_bindings = kbm.registry.key_bindings
+class _DummyTerminal(object):
+    """Used as a buffer to get prompt_toolkit bindings
+    """
+    handle_return = None
+    input_splitter = None
+    display_completions = None
+
+ipy_bindings = create_ipython_shortcuts(_DummyTerminal()).bindings
 
 dummy_docs = []  # ignore bindings without proper documentation
 
